internal/pkg/remote/client.go

```diff
@@ -220,13 +220,30 @@ func (c *Client) Send(
 			c.log.With("error", err).Debugf(msg)
 			continue
 		}
+		c.checkApiVersionHeaders(reqID, resp)
 
 		return resp, nil
 	}
 
 	return nil, fmt.Errorf("all hosts failed: %w", multiErr)
 }
 
+func (c *Client) checkApiVersionHeaders(reqID string, resp *http.Response) {
```

Comment:
Would it be possible to write a unit test for this method with cases for when the warning header is set and not set?

Reply:
This is an internal implementation detail that is subject to change due to refactoring. I added unit tests for the client public function `Send` about the api version and the warning headers above in the `TestElasticApiVersion` where we assert the behavior of the client with a test http server :thinking: